sherpa/astro/tests/test_astro_data_asca_unit.py

```diff
@@ -0,0 +1,869 @@
+
+    assert rmf.f_chan.size == nsize1
+    assert rmf.f_chan.min() == 14
+    assert rmf.f_chan.max() == 19
+    assert rmf.f_chan.sum() == 16711
+
+    assert rmf.n_chan.size == nsize1
+    assert rmf.n_chan.min() == 1
+    assert rmf.n_chan.max() == 426
+    assert rmf.n_chan.sum() == nsize2
+
+    assert rmf.matrix.size == nsize2
+    assert rmf.matrix.min() == pytest.approx(1.0003149952808599e-07)
+    assert rmf.matrix.max() == pytest.approx(0.11968599259853363)
+    assert rmf.matrix.sum() == pytest.approx(544.8837793875416)
+    assert np.argmax(rmf.matrix) == 4307
+
+    for field in ['e_min', 'e_max']:
+        assert getattr(rmf, field).size == nchans
+
+    # Expect the upper edge of bin j to equal the lower
+    # edge of bin j + 1. Note that we check that the low
+    # bin has been relpaced (i.e. is not 0.0).
+    #
+    assert rmf.energ_lo[1:] == pytest.approx(rmf.energ_hi[:-1])
+    assert rmf.energ_lo[0] == pytest.approx(0.20000000298023)
+    assert rmf.energ_hi[0] == pytest.approx(0.21000000834465)
+    assert rmf.energ_lo[-1] == pytest.approx(11.9901790619)
+    assert rmf.energ_hi[-1] == pytest.approx(12.0)
+
+    assert rmf.e_min[1:] == pytest.approx(rmf.e_max[:-1])
+    assert rmf.e_min[0] == pytest.approx(0.013292968273162842)
+    assert rmf.e_max[0] == pytest.approx(0.04193559288978577)
+    assert rmf.e_min[-1] == pytest.approx(14.649674415588379)
+    assert rmf.e_max[-1] == pytest.approx(14.678317070007324)
+
+
+def check_rmf1(rmf):
+
+    assert isinstance(rmf, DataRMF)
+
+    nchans = 1024
+    nbins = 1180
+    nsize1 = 1171
+    nsize2 = 505483
+
+    assert rmf.detchans == nchans
+    assert rmf.offset == 1
+
+    for field in ['energ_lo', 'energ_hi']:
+        assert getattr(rmf, field).size == nbins
+
+    assert rmf.n_grp.size == nbins
+    assert rmf.n_grp.min() == 0
+    assert rmf.n_grp.max() == 1
+    assert rmf.n_grp.sum() == nsize1
+
+    assert rmf.f_chan.size == nsize1
+    assert rmf.f_chan.min() == 26
+    assert rmf.f_chan.max() == 26
+    assert rmf.f_chan.sum() == 30446
+
+    assert rmf.n_chan.size == nsize1
+    assert rmf.n_chan.min() == 1
+    assert rmf.n_chan.max() == 851
+    assert rmf.n_chan.sum() == nsize2
+
+    assert rmf.matrix.size == nsize2
+    # TODO: asking for the min value causes this test to hang on my
+    # machine. I can read in the file outside of the test and get
+    # the minimum value. Why is it just here?
+    #
+    # assert rmf.matrix.min() == pytest.approx(1.0003149952808599e-7)
+    assert (rmf.matrix > 1e-7).all()  # approximate the above
+    assert rmf.matrix.max() == pytest.approx(0.11497905850410461)
+    assert rmf.matrix.sum() == pytest.approx(552.932040504181)
+    assert np.argmax(rmf.matrix) == 7324
+
+    for field in ['e_min', 'e_max']:
+        assert getattr(rmf, field).size == nchans
+
+    # Expect the upper edge of bin j to equal the lower
+    # edge of bin j + 1. Note that we check that the low
+    # bin has been relpaced (i.e. is not 0.0).
```

Comment:
replaced

Reply:
gah - the same copy-n-paste error from #2118